src/openstudio_lib/YearSettingsWidget.cpp

```diff
@@ -510,55 +497,65 @@ void YearSettingsWidget::onDstEndDayWeekMonthChanged() {
   std::string s_dayOfWeek;
   std::string s_monthOfYear;
 
-  int i_nthDayOfWeekInMonth = m_endWeekBox->currentIndex();
-  s_dayOfWeek = m_endDayBox->currentText().toStdString();
-  s_monthOfYear = m_endMonthBox->currentText().toStdString();
+  NthDayOfWeekInMonth _nth(m_endWeekBox->currentIndex() + 1); // NthDayOfWeekInMonth starts at 1...
 
-  NthDayOfWeekInMonth _nth(i_nthDayOfWeekInMonth + 1);
-  DayOfWeek _dayOfWeek = dayOfWeek(s_dayOfWeek);
-  MonthOfYear _monthOfYear = monthOfYear(s_monthOfYear);
+  DayOfWeek _dayOfWeek(m_endDayBox->currentIndex());
+  MonthOfYear _monthOfYear(m_endMonthBox->currentIndex() + 1); // MonthOfYear starts at 1...
 
   emit dstEndDayOfWeekAndMonthChanged(_nth.value(), _dayOfWeek.value(), _monthOfYear.value());
 }
 
-std::vector<std::string> YearSettingsWidget::weeksInMonth() {
-  std::vector<std::string> result(5);
-
-  result[0] = "First";
-  result[1] = "Second";
-  result[2] = "Third";
-  result[3] = "Fourth";
-  result[4] = "Last";
-
+QStringList YearSettingsWidget::weeksInMonth() {
+  QStringList result = {tr("First"), tr("Second"), tr("Third"), tr("Fourth"), tr("Last")};
   return result;
 }
 
-std::vector<std::string> YearSettingsWidget::daysOfWeek() {
-  std::vector<std::string> result;
+QStringList YearSettingsWidget::validDayofWeekforStartDay() {
 
-  std::map<int, std::string> nameMap = DayOfWeek::getNames();
-
-  for (auto it = nameMap.begin(); it != nameMap.end(); ++it) {
-    result.push_back(it->second);
-  }
+  // Similar to model::YearDescription::validDayofWeekforStartDayValues(), but we need to be able to translate them
+  QStringList result  = {
```

Comment:
Translating from strings in the IDD is going to be expensive.  I wonder if we could write a script to export all the object names and enums from the IDD, translate that, then just feed `tr` the strings from the IDD?

Reply:
We could use https://translate.google.com/?sl=auto&tl=en&op=docs to do a head start?  Or maybe people prefer to not translate the IDD items?

Reply:
I'd say generally I'm going to be opposed to translating the IDD Enums. For days and months it makes a ton of sense, but otherwise not so much. It's not like you're going to find the E+ I/O reference guide in anything else than English, so I'd want to keep that aligned.